pkg/providers/amifamily/resolver.go

```diff
@@ -122,16 +123,24 @@ func NewResolver(amiProvider Provider) *Resolver {
 // Multiple ResolvedTemplates are returned based on the instanceTypes passed in to support special AMIs for certain instance types like GPUs.
 func (r Resolver) Resolve(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType, capacityType string, options *Options) ([]*LaunchTemplate, error) {
 	amiFamily := GetAMIFamily(nodeClass.Spec.AMIFamily, options)
-	amis, err := r.amiProvider.Get(ctx, nodeClass, options)
+	if len(nodeClass.Status.AMIs) == 0 {
+		return nil, fmt.Errorf("no amis exist given constraints")
+	}
+	amis, err := r.amiProvider.Get(ctx, nodeClass)
```

Comment:
Why do we need another Get() if we already have the NodeClass and AMIs in the status above?